moq-transport/src/serve/stream.rs

```diff
@@ -303,7 +303,7 @@ impl StreamGroupReader {
         loop {
             {
                 let state = self.state.lock();
-                if self.index < state.objects.len() {
+                if self.index < state.objects.len() - 1 {
```

Comment:
I believe rearranging the inequality like `if self.index + 1 < state.objects.len() {` could be safer in most cases, considering that `state.objects.len()` can return `0`.